tiatoolbox/tools/registration/prealignment.py

```diff
@@ -0,0 +1,89 @@
+import cv2
+import numpy as np
+import scipy.ndimage as ndi
+from skimage import exposure
+from skimage.util import img_as_float
+
+from tiatoolbox.utils.metrics import dice
+
+
+def prealignment(fixed_img, moving_img, fixed_mask, moving_mask, rotation_steps=10):
+    """This function performs coarse registration of a pair of images
+
+    Args:
+        fixed_img (:class:`numpy.ndarray`):
+            A grayscale fixed image
+        moving_img (:class:`numpy.ndarray`):
+            A grayscale moving image
+        fixed_mask (:class:`numpy.ndarray`):
+            A binary tissue mask for the fixed image
+        moving_mask (:class:`numpy.ndarray`):
+            A binary tissue mask for the moving image
+
+    Returns:
+        :class:`numpy.ndarray`:
+            A transform matrix
+
+    """
+    if len(fixed_mask.shape) != 2:
+        fixed_mask = fixed_mask[:, :, 0]
+    if len(moving_mask.shape) != 2:
+        moving_mask = moving_mask[:, :, 0]
+
+    fixed_mask, moving_mask = np.uint8(fixed_mask > 0), np.uint8(moving_mask > 0)
+
+    if len(np.unique(fixed_mask)) == 1 or len(np.unique(fixed_mask)) == 1:
+        raise ValueError(f'{"The foreground is missing in the mask."}')
+
+    if len(fixed_img.shape) != 2 or len(moving_img.shape) != 2:
+        raise ValueError(f'{"The input images should be grayscale images."}')
+
+    if fixed_img.shape != fixed_img.shape or moving_img.shape != moving_mask.shape:
```

Comment:
```suggestion
    if fixed_img.shape != fixed_mask.shape or moving_img.shape != moving_mask.shape:
```

Reply:
Fixed it https://github.com/TissueImageAnalytics/tiatoolbox/blob/dev-prealignment/tiatoolbox/tools/registration/prealignment.py#:~:text=if%20fixed_img.,)